internal/datastore/crdb/pool/pool.go

```diff
@@ -67,6 +69,11 @@ func NewRetryPool(ctx context.Context, name string, config *pgxpool.Config, heal
 		delete(p.gc, conn)
 
 		healthTracker.SetNodeHealth(nodeID(conn), true)
+
+		if err := limiter.Wait(ctx); err != nil {
```

Comment:
let's say there is a network partition and all connections fail, and immediately partition goes away and tries to recover. If the connection pool was emptied because all connections were in bad, SpiceDB will take the hit of the workload queuing up but it would be gentle with the database. I think that seems reasonable and seems better than making SpiceDB saturated and the database too - may be able to recover faster. But it might be important to consider shed load on the SpiceDB API. This is anyway a problem today that is indistinguishable from normal pool contention.